src/ucs/config/parser.c

```diff
@@ -1851,6 +1852,64 @@ void ucs_config_parser_print_all_opts(FILE *stream, const char *prefix,
     }
 }
 
+static void ucs_config_parser_concat_fuzzy_matches_recurs(
+        const ucs_config_field_t *fields, const char *env_prefix,
+        const char *table_prefix, const char *user_var,
+        ucs_string_buffer_t *fuzzy_buff)
+{
+    char ucx_var[128]                 = {0};
+    const uint32_t MAX_FUZZY_DISTANCE = 3;
+    const ucs_config_field_t *field;
+    uint32_t distance;
+
+    for (field = fields; !ucs_config_field_is_last(field); ++field) {
+        if (ucs_config_is_table_field(field)) {
+            ucs_config_parser_concat_fuzzy_matches_recurs(
+                    field->parser.arg, env_prefix, table_prefix, user_var,
+                    fuzzy_buff);
+        }
+
+        snprintf(ucx_var, sizeof(ucx_var), "%s%s%s", env_prefix,
+                 table_prefix ? table_prefix : "", field->name);
+
+        distance = ucs_levenshtein(user_var, ucx_var);
+        if ((distance > 0) && (distance <= MAX_FUZZY_DISTANCE)) {
```

Comment:
I cannot understand why it is 2/3 chars because as I understand the difference is only 2 points according to the Levenstein metric. But anyway I agree that this cannot be expressed by the ratio based logic. So let's just leave 3 misspells as a threshold.